lib/arena.rb

```diff
@@ -1,2 +1,93 @@
-class Arena
+# require "pry"
+
+require_relative "gladiator"
+
+class Arena < Gladiator
+  attr_accessor :name, :gladiators
+
+ # 3 gladiators starts off empty
+
+  def initialize(name)
+    @name = name.capitalize
+    @gladiators = []
+    @counter = 0
+  end
+
+# 1 Arena name
+# 2 Arena name is always capitalized
+  def name
+    @name
+  end
+
+# 4 add_gladiator adds a gladiator to the arena
+# 5 gladiator can add multiple gladiators to the arena
+# 6 but not more than two at the same time
+
+
+
+def add_gladiator(gladiator)
+  if @gladiators.length == 2
+    return
+  else
+    @gladiators << gladiator
+  end
 end
+
+# 7 fight when no gladiators does nothing
+# 8 fight when one gladiator does nothing
+
+#9 when Spear v Trident kills: spear when fighting trident
+#10 when Club v Spear kills: club when fighting spear
+#11 when Trident v Club kills: trident when fighting club
+
+#
+
+def fight
+  if @gladiators.length == 2
+    # puts "#{@gladiators}"
+    @gladiators.each do |type|
+      weapons_rank = ["Spear", "Club", "Trident"]
```

Comment:
I think you're on the right track here re: drying up the code. maybe you could compare indexes of the weapons for each gladiator. e.g.

Want to pair on this one?